app/controllers/replies_controller.rb

```diff
@@ -297,19 +306,153 @@ def require_edit_permission
     redirect_to post_path(@reply.post)
   end
 
-  def preview(written)
-    @written = written
-    @post = @written.post
+  def get_multi_replies
+    # Get the multi replies stored in the page JSON
+    @multi_replies_params = JSON.parse(params.fetch(:multi_replies_json, "[]")).map do |reply_json|
+      permitted_params(ActionController::Parameters.new({ reply: reply_json }), [:id])
+    end
+    @multi_replies = @multi_replies_params.map do |reply_params|
+      Reply.new(reply_params).tap { |r| r.user = current_user }
+    end
+  end
+
+  def preview_reply(reply)
+    # Previewing a specific reply
+    @post = reply.post
+    @written = reply
     @written.user = current_user unless @written.user
     @audits = @written.id.present? ? { @written.id => @written.audits.count } : {}
+    @reply = @written # So that editor_setup shows the correct characters
+    preview_replies
+  end
 
+  def add_to_multi_reply(reply, reply_params)
+    # Adding a new reply to a multi reply
+    post_id = params[:reply][:post_id]
+    ReplyDraft.draft_for(post_id, current_user.id)&.destroy!
+
+    # Save NPC
+    if reply.character&.new_record?
+      if reply.character.save
+        flash[:success] = "Your new NPC has been persisted!"
+        params[:reply][:character_id] = reply.character.id
+        reply_params[:character_id] = reply.character.id
+        reply.character_id = reply.character.id
+      else
+        flash[:error] = "There was a problem persisting your new NPC."
+      end
+    end
+
+    # Set up editor
+    @post = reply.post
+    empty_reply_hash = permitted_params.permit(:character_id, :icon_id, :character_alias_id)
+    @empty_written = @post.build_new_reply_for(current_user, empty_reply_hash)
+    @empty_written.editor_mode = reply.editor_mode
+    @reply = @empty_written # So that editor_setup shows the correct characters
```

Comment:
Huh, what's the deal with this?

Reply:
This is so that the editor will always show no reply + the prior multi reply's character, the way the regular editor at the bottom of a thread's page does

Reply:
where's the `@empty_written` being used distinct from the `@reply`?

Reply:
`@reply` is used by the `editor_setup` function, `@empty_written` is used by the `_preview` partial

Reply:
...not partial. just page.

Reply:
Having looked at that earlier I'm not sure why you didn't just use `@reply` when adding a variable to it but ok

Reply:
'cause I do different things depending on whether `@written` is present but I guess I could just pass a boolean maybe?